### PR TITLE
[1/N refactor NCCLPG error logs] Make NCCL PG error messages more accurate and simpler

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #134036
* __->__ #134017

We did a thorough review on all the error messages we are logging inside PGNCCL, and we want to make log message simpler and more accurate, this is the first PR for this effort.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 1]
```diff
@@ -1742,8 +1740,9 @@ void ProcessGroupNCCL::watchdogHandler() {
                 reinterpret_cast<uint8_t*>(&rank) + sizeof(rank));
             globalStore_->set(std::string(EXCEPTION_DUMP), vec);
             if (!shouldDump_.load()) {
-              LOG(ERROR) << logPrefix()
-                         << "First watchdog to set the dump signal.";
+              LOG(ERROR)
+                  << logPrefix()
+                  << "First watchdog to set the dump signal to TCPStore.";
```
[the remark this answers is not in the document]
Comment (fduwjj) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1745]: I think it's other ranks, but I agree with the broadcasting part.

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 2]
```diff
@@ -1732,8 +1729,9 @@ void ProcessGroupNCCL::watchdogHandler() {
             ", last completed NCCL work: ",
             pgStatus_->lastCompletedSeq,
             ".");
-        // try to dump flight records if exception happens.
-        // Flight recorder behavior should be independent of desync Debug
+        // try to notify other ranks via global TCPStore to dump the flight
+        // recorder if exception happens. Flight recorder behavior is
+        // independent of desync Debug.
         if (dumpOnException_) {
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1735]: nit: some folks got confused by 'exception' here, iirc.  it is actually timeout or exception, isn't it?

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 3]
```diff
@@ -1742,8 +1740,9 @@ void ProcessGroupNCCL::watchdogHandler() {
                 reinterpret_cast<uint8_t*>(&rank) + sizeof(rank));
             globalStore_->set(std::string(EXCEPTION_DUMP), vec);
             if (!shouldDump_.load()) {
-              LOG(ERROR) << logPrefix()
-                         << "First watchdog to set the dump signal.";
+              LOG(ERROR)
+                  << logPrefix()
+                  << "Broadcasting the dump signal of flight-recorder to other processes via TCPStore.";
```
Comment (wconstab) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1745]: nit: "flight-recorder dump signal" seems clearer than "dump signal of flight-recorder"

**Review by wconstab — APPROVED** [review]

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp (hunk 2 above)
Comment (fduwjj) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1735]: Sure, I also changed the variable name, lol

**Pull request closed without merging**